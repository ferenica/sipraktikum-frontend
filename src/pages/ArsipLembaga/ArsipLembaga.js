import React, {Component} from 'react';
import Navbar from '../../components/Navbar/Navbar';
import PropTypes from 'prop-types';

import styled from 'styled-components';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import ArsipLembagaSearch from './ArsipLembagaSearch/ArsipLembagaSearch';

// Styled components
const HeaderContainer = styled.div`
position: sticky;
top: 0;
z-index: -1;
margin-bottom: -10vw;
`;

/**
 * Arsip Lembaga page
 */

class ArsipLembaga extends Component {
    newDate = new Date();  
    year = this.newDate.getFullYear() - 6;

    /**
   * @return {*} page
   */
    render (){
        const StyledContainer = styled(Container)({
            paddingTop: '2%',
            minHeight: '100vh',
            backgroundColor: '#FFFFFF',
            marginTop: '3vh',
            marginBottom: '2.5rem',
            borderRadius: 10,
            boxShadow:
            `6px 0px 8px 0 rgba(0, 0, 0, 0.05),
            6px 0px 6px 0 rgba(0, 0, 0, 0.03),
            -6px 0px 8px 0 rgba(0, 0, 0, 0.05),
            -6px 0px 6px 0 rgba(0, 0, 0, 0.03)`,
            width: '1000px',
        });

        return (
            <>
                <Navbar
                    isAuthenticated={this.props.isAuthenticated}
                    isAdmin={this.props.isAdmin}
                    isDosen={this.props.isDosen}
                />
                <HeaderContainer>
                    <img
                        style={{width: '100vw'}}
                        src={require('../../assets/MKPPL/Orange_Header.png')}
                        alt="header"
                    />
                </HeaderContainer>
                <StyledContainer>
                    <div style={{marginLeft: '4%'}}>
                        <h1
                        className="judul"
                        style={{color: '#000000', fontFamily: 'Nunito Sans'}}>
                            Arsip Mitra Lembaga
                        </h1>
                        <Typography
                            component="div"
                            variant="body1"
                            display="block"
                            gutterBottom>
                            <center>Daftar lembaga yang  bekerjasama dengan Departemen Ilmu Kesejahteraan Sosial</center>
                        </Typography>
                        <Typography
                            component="div"
                            variant="body1"
                            display="block"
                            gutterBottom>
                            <center>tahun {this.year} dan sebelumnya</center>
                        </Typography>
                        <br></br><br></br>
                    </div>
                    <ArsipLembagaSearch isAdmin={this.props.isAdmin} isDosen={this.props.isDosen}/>
                </StyledContainer>
            </>
        );  
    }
}

export default ArsipLembaga;

ArsipLembaga.propTypes = {
    isAuthenticated: PropTypes.bool,
    isAdmin: PropTypes.bool,
    isDosen: PropTypes.bool,
};