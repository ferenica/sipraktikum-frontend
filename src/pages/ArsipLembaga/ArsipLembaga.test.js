import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import ArsipLembaga from './ArsipLembaga';

Enzyme.configure({adapter: new EnzymeAdapter()});
const setup = (props = {}, state = null) => {
  const wrapper = shallow(<ArsipLembaga {...props} />);
  if (state) wrapper.setState(state);
  return wrapper;
};

describe('ArsipLembaga', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('exist navbar', () => {
    const navbar = wrapper.find('Navbar');
    expect(navbar.length).toBe(1);
  });

  it('exist ArsipLembagaSearch', () => {
    const content = wrapper.find('ArsipLembagaSearch');
    expect(content.length).toBe(1);
  });

  it('should renders correctly', () => {
    const component = shallow(<ArsipLembaga />);
    expect(component).toMatchSnapshot();
  });
});