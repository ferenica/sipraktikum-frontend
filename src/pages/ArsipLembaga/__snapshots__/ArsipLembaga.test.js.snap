// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArsipLembaga should renders correctly 1`] = `ShallowWrapper {}`;
